src/ctxt.ml

```diff
@@ -3,37 +3,64 @@
 open Extra
 open Terms
 
-(** Representation of a typing context, associating a type (or [Term.term]) to
-    free [Bindlib] variables. *)
-type t = (tvar * term) list
+type t = ctxt
 
 (** [empty] is the empty context. *)
 let empty : t = []
 
-(** [add x a ctx] maps the variable [x] to the type [a] in [ctx]. *)
-let add : tvar -> term -> t -> t =
-  fun x a ctx -> (x,a)::ctx
+(** [assume h ctx] adds list of assumptions [h] mapping variables to their
+    types to context [ctx]. *)
+let assume : (tvar * term) list -> t -> t = fun hyps ctx ->
+  List.map (fun (x,a) -> (x, a, None)) hyps @ ctx
 
-(** [unbind_ctxt ctx a b] returns the triple [(x,b,ctx')] such that [(x,b)]
-   is the unbinding of [b] and [ctx'] is the context [ctx] extended with
-   [(x,a)] if [x] occurs in [b]. *)
-let unbind ctx a b =
+(** [define x a t ctx] maps variable [x] to term [t] of type [a] in [ctx]. *)
+let define : tvar -> term -> term -> t -> t = fun x a t ctx ->
+  (x, a, Some(t)) :: ctx
+
```

Comment:
why not having a single function `add` with an option argument?

Reply:
I have removed add functions